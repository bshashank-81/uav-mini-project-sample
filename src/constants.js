export const USERNAME_LABEL = "Username";
export const USERNAME_PLACEHOLDER = "Enter username";
export const PASSWORD_LABEL = "Password";
export const PASSWORD_PLACEHOLDER = "Enter password";
export const SIGN_IN = "Sign in";
export const INVALID_CREDENTIALS = "invalid credentials";
export const NETWORK_ERROR = "network error";
export const LOGIN = "Login";
export const MAPS = "Maps";
export const LOCATIONS = "Locations";
export const LATITUDE = "LATITUDE";
export const LONGITUDE = "LONGITUDE";
export const ID = "ID";
export const NEW_LOCATION = "Add Location";
export const LOGOUT = "Logout";
