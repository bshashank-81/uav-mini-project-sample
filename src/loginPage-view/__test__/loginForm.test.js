import React from "react";
import { shallow } from "enzyme";
import LoginForm from "../components/loginForm";
import * as Constants from "../../constants";
import { render, fireEvent, act, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import fetchMock from "fetch-mock";
import { BrowserRouter as Router, MemoryRouter } from "react-router-dom";
import { create } from "react-test-renderer";
import { createMemoryHistory } from "history";
import { createBrowserHistory } from "history";
import { Provider } from "react-redux";

describe("Login Form component", () => {
  let userProps = {
    label: "Username",
    placeholder: "Enter email",
    username: "admin",
    onChange: jest.fn(),
  };
  let passwordProps = {
    label: "Password",
    placeholder: "Enter password",
    password: "zemoso@123",
    onChange: jest.fn(),
  };
  let container;
  const wrapper = shallow(<LoginForm />);
  beforeEach(() => {
    container = document.createElement("LoginForm");
    document.body.appendChild(container);
  });
  test("Matches the snapshot", () => {
    const loginForm = create(
      <Router>
        <LoginForm />
      </Router>
    );
    expect(loginForm.toJSON()).toMatchSnapshot();
  });
  it("label should be Username", () => {
    const content = wrapper.find({ "data-testid": "usernameLabel" }).text();
    expect(content).toBe("Username");
  });
  it("should have password label", () => {
    const content = wrapper.find({ "data-testid": "passwordLabel" }).text();
    expect(content).toBe("Password");
  });
  it("should have Sign in content on the button", () => {
    const content = wrapper.find({ "data-testid": "signIn" }).text();
    expect(content).toBe("Sign in");
  });
  test("handle click event should work", async () => {
    const { getByTestId } = render(
      <Router>
        <LoginForm />
      </Router>
    );
    fireEvent.click(getByTestId("signIn"));
    fetchMock.mock("http://localhost:3001/login", 200);
    const response = await fetch("http://localhost:3001/login");
    expect(response.ok).toBeTruthy;
  });
  test("error message must be displayed for wrong inputs", async () => {
    global.fetch = jest.fn().mockImplementation(() => {
      var p = new Promise((resolve, reject) => {
        resolve({
          ok: false,
        });
      });
      return p;
    });
    const { getByTestId, getByRole, getByText } = render(
      <Router>
        <LoginForm />
      </Router>
    );
    fireEvent.change(getByTestId("username"), {
      target: { value: "admin" },
    });
    fireEvent.change(getByTestId("password"), {
      target: { value: "zemoso" },
    });
    await fireEvent.click(getByRole("button"));
    expect(getByTestId("error")).toBeInTheDocument();
    expect(getByText("invalid credentials")).toBeInTheDocument();
  });
  test("should login on entering correct credentials", async () => {
    global.fetch = jest.fn().mockImplementation(() => {
      var x = new Promise((resolve, reject) => {
        resolve({
          ok: true,
          json: function () {
            return { access_token: "fake_access_token" };
          },
        });
      });
      return x;
    });
    const { getByTestId, getByRole } = render(
      <Router>
        <LoginForm />
      </Router>
    );
    fireEvent.change(getByTestId("username"), {
      target: { value: "admin" },
    });
    fireEvent.change(getByTestId("password"), {
      target: { value: "zemoso@123" },
    });
    await act(() => fireEvent.click(getByRole("button")));
  });
});
