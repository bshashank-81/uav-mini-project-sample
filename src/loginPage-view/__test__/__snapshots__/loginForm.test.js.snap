// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form component Matches the snapshot 1`] = `
<form
  className="makeStyles-form-1"
>
  <div
    className="MuiFormControl-root makeStyles-textField-4 MuiFormControl-fullWidth"
  >
    <p
      className="MuiTypography-root makeStyles-label-3 MuiTypography-body1"
      data-testid="usernameLabel"
    >
      Username
    </p>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          data-testid="username"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Enter username"
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="MuiFormControl-root makeStyles-textField-4 MuiFormControl-fullWidth"
  >
    <p
      className="MuiTypography-root makeStyles-label-3 MuiTypography-body1"
      data-testid="passwordLabel"
    >
      Password
    </p>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          data-testid="password"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Enter password"
          required={true}
          type="Text"
          value=""
        />
      </div>
    </div>
  </div>
  <p
    className="MuiTypography-root makeStyles-error-2 MuiTypography-body1"
    data-testid="error"
    id="error"
  />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-5 MuiButton-containedPrimary MuiButton-fullWidth"
    data-testid="signIn"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Sign in
    </span>
  </button>
</form>
`;
